### PR TITLE
Add setup and teardown logic to torchhub tests. Also cleaned up uses of temp folders

### PR DESCRIPTION
**Pull request by NicolasHug**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #74431
* #74430
* #74429
* #74428
* __->__ #74427

Differential Revision: [D35011832](https://our.internmc.facebook.com/intern/diff/D35011832)

### CODE REVIEW

**Review by vmoens — APPROVED** [review]
LGTM, thanks!

**Comment by NicolasHug** [comment]
@NicolasHug has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35011832).

**Comment by NicolasHug** [comment]
The diff was landed and the changes are included in the `master` branch

https://github.com/pytorch/pytorch/blob/6da9a79e289ec3eed2d12d067c296e6d023d28e3/test/test_utils.py#L594-L601

Not sure why this wasn't automatically closed

File: test/test_utils.py [hunk 1]
```diff
@@ -589,6 +589,16 @@ def sum_of_state_dict(state_dict):
 
 @unittest.skipIf(IS_SANDCASTLE, 'Sandcastle cannot ping external')
 class TestHub(TestCase):
+
+    def setUp(self):
+        self.previous_hub_dir = torch.hub.get_dir()
+        self.tmpdir = tempfile.TemporaryDirectory('hub_dir')
+        torch.hub.set_dir(self.tmpdir.name)
```
Comment (janeyx99) [inline test/test_utils.py:596]: this should call super().setUp()!

File: test/test_utils.py [hunk 2]
```diff
@@ -589,6 +589,16 @@ def sum_of_state_dict(state_dict):
...
+
+    def tearDown(self):
+        torch.hub.set_dir(self.previous_hub_dir)  # probably not needed, but can't hurt
```
Comment (janeyx99) [inline test/test_utils.py:599]: this should call super().tearDown()!

**Pull request closed without merging**